File: 205/solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        relations1 = {}
        relations2 = {}
        n = len(s)
        for i in range(n):
            x = s[i]
            y = t[i]
            if x in relations1:
                if relations1[x] != y:
                    return False
            elif y in relations2:
                if relations2[y] != x:
                    return False
            else:
                relations1[x] = y
                relations2[y] = x
        return True

File: 205/Solution.java
import java.util.HashMap;

class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> relations1 = new HashMap<>();
        HashMap<Character, Character> relations2 = new HashMap<>();
        int n = s.length();
        for (int i = 0; i < n; i++) {
            char x = s.charAt(i);
            char y = t.charAt(i);
            if (relations1.containsKey(x)) {
                if (relations1.get(x) != y) {
                    return false;
                }
            } else if (relations2.containsKey(y)) {
                if (relations2.get(y) != x) {
                    return false;
                }
            } else {
                relations1.put(x, y);
                relations2.put(y, x);
            }
        }
        return true;
    }
}